hibernate-core/src/main/java/org/hibernate/query/internal/ParameterMetadataImpl.java

```diff
@@ -42,19 +43,19 @@ public class ParameterMetadataImpl implements ParameterMetadataImplementor {
 	public static final ParameterMetadataImpl EMPTY = new ParameterMetadataImpl();
 
 	private final Map<QueryParameterImplementor<?>, List<SqmParameter<?>>> queryParameters;
-
-	private final Set<String> names;
-	private final Set<Integer> labels;
-
+	private final Map<String, QueryParameterImplementor<?>> queryParametersByNames;
+	private final Map<Integer, QueryParameterImplementor<?>> queryParametersByLabels;
```

Comment:
I'd prefer something like `queryParametersByName` and `queryParametersByPosition`.